src/firmware/guest/mod.rs

```diff
@@ -85,22 +86,21 @@ impl Firmware {
         &mut self,
         message_version: Option<u8>,
         ext_report_request: SnpExtReportReq,
-    ) -> Result<SnpReportRsp, Indeterminate<Error>> {
+    ) -> Result<(AttestationReport, CertTable), Indeterminate<Error>> {
         let ext_report_response: SnpReportRsp = SnpReportRsp::default();
+        let mut cert_table: CertTable = Default::default();
 
         SNP_GET_EXT_REPORT.ioctl(
             &mut self.0,
-            &mut SnpGuestRequest::new(
-                message_version,
-                &SnpExtReportReq {
-                    data: ext_report_request.data,
-                    certs_address: ext_report_request.certs_address,
-                    certs_len: ext_report_request.certs_len,
-                },
-                &ext_report_response,
-            ),
+            &mut SnpGuestRequest::new(message_version, &ext_report_request, &ext_report_response),
         )?;
 
-        Ok(ext_report_response)
+        if let Some(linux_cert_table) =
+            unsafe { (ext_report_request.certs_address as *const FFICertTable).as_ref() }
+        {
+            cert_table = linux_cert_table.to_uapi();
```

Comment:
Here is where a `UapiCertTableEntry` will be returned, but some other `Vec<Certificate>` should be returned from the `CertTableEntry` entries.

Reply:
If you review my comment above, the purpose of the `to_uapi()` function is to parse the certificates into the Rust-friendly structure provided.